doc/collision_detection.md

```diff
@@ -0,0 +1,212 @@
+with it. If you so want it to be notified, then you need to extend it and implement the
+`collisionCallback` method for it.
+
+## Shapes
+### Shape
+A Shape is the base class for representing a scalable geometrical shape. The shapes have different
+ways of defining how they look, but they all have a size and angle that can be modified and the
+shape definition will scale or rotate the shape accordingly.
+
+There are currently three shapes: [Polygon}(#Polygon), [Rectangle](#Rectangle) and
+[Cirlce](#Circle).
+
+### HitboxShape
+A HitboxShape is a Shape defined from the center position of the component that it is attached to
+and it has the same bounding size and angle as the component. You can set `localPosition` to have
+the position of the shape deviate from the center of the component. A HitboxShape is the type of
+shape that you add to your Hitbox, or Collidable. Usually these types of shapes are the only ones
+that you need to use.
+
+#### HitboxPolygon
+It should be noted that if you want to use collision detection or `containsPoint` on the `Polygon`,
+the polygon needs to be convex. So always use convex polygons or you will most likely run into
+problems if you don't really know what you are doing. It should also be noted that you should always
+define the vertices in your polygon in a clockwise order.
+
+In comparision to the normal `Polygon`, there is only one way to create a `HitboxPolygon`, the only
+mandatory argument is a list of `Vector2` which defines how your polygon should look, but not the
+size or position for it (since that sill be defined by the component that you attach it to).
+For example you could create a diamond like in the [Collidable](#Collidable) example like this:
+
+```dart
+HitboxPolygon([
+  Vector2(0, 1),  // Middle of top wall
+  Vector2(1, 0),  // Middle of right wall
+  Vector2(0, -1), // Middle of bottom wall
+  Vector2(-1, 0), // Middle of left wall
+]);
+```
+
+The other hitbox shapes don't have any mandatory constructor, but that is because they can have a
+sane default calculated from the size of the collidable that they are attached to, but since a
+polygon can be made in an infinite number of ways inside of a bounding box you have to add the
+definition in the constructor for this shape.
+
+The vectors in the example defines percentages of the length from the center to the edge of the
+screen in both x and y axis, so for our first item in our list (`Vector2(0, 1)`) we are pointing on
+on the middle of the top wall of the bounding box, since the cordinate system here is defined from
+the center of the polygon.
+
+![An example of how to define a polygon shape](images/polygon_shape.png)
+
+In the image you can see how the polygon shape formed by the purple arrows is defined by the red
+arrows.
+
+
+#### HitboxRectangle
+A `HitboxRectangle` is really just a simplified Polygon, but it can be defined more easily.
```

Comment:
I imagine it's also a more performatic implementation? or does it use Polygon under the hood?